pkg/util/util.go

```diff
@@ -190,3 +200,106 @@ func TrimDuplicatedSpace(s string) string {
 	s = reg.ReplaceAllString(s, " ")
 	return s
 }
+
+type EXEC interface {
+	RunCommand(string) (string, error)
+}
+
+type ExecCommand struct {
+}
+
+func (ec *ExecCommand) RunCommand(cmd string) (string, error) {
+	out, err := exec.Command("sh", "-c", cmd).CombinedOutput()
+	return string(out), err
+}
+
+type AzcopyFunc struct {
+	ExecCmd EXEC
+}
+
+// GetAzcopyJob get the azcopy job status if job existed
+func (af *AzcopyFunc) GetAzcopyJob(dstBlobContainer string) (AzcopyJobState, string, error) {
+	cmdStr := fmt.Sprintf("azcopy jobs list | grep %s -B 3", dstBlobContainer)
+	// cmd output example:
+	// JobId: ed1c3833-eaff-fe42-71d7-513fb065a9d9
+	// Start Time: Monday, 07-Aug-23 03:29:54 UTC
+	// Status: Completed (or Cancelled, InProgress)
+	// Command: copy https://{accountName}.file.core.windows.net/{srcFileshare}{SAStoken} https://{accountName}.file.core.windows.net/{dstFileshare}{SAStoken} --recursive --check-length=false
+	// --
+	// JobId: b598cce3-9aa9-9640-7793-c2bf3c385a9a
+	// Start Time: Wednesday, 09-Aug-23 09:09:03 UTC
+	// Status: Cancelled
+	// Command: copy https://{accountName}.file.core.windows.net/{srcFileshare}{SAStoken} https://{accountName}.file.core.windows.net/{dstFileshare}{SAStoken} --recursive --check-length=false
+	if af.ExecCmd == nil {
+		af.ExecCmd = &ExecCommand{}
+	}
+	out, err := af.ExecCmd.RunCommand(cmdStr)
+	// if grep command returns nothing, the exec will return exit status 1 error, so filter this error
+	if err != nil && err.Error() != "exit status 1" {
+		klog.Warningf("failed to get azcopy job with error: %v, jobState: %v", err, AzcopyJobError)
+		return AzcopyJobError, "", fmt.Errorf("couldn't list jobs in azcopy %v", err)
+	}
+	jobid, jobState, err := parseAzcopyJobList(out, dstBlobContainer)
+	if err != nil || jobState == AzcopyJobError {
+		klog.Warningf("failed to get azcopy job with error: %v, jobState: %v", err, jobState)
+		return AzcopyJobError, "", fmt.Errorf("couldn't parse azcopy job list in azcopy %v", err)
+	}
+	if jobState == AzcopyJobCompleted {
+		return jobState, "100.0", err
+	}
+	if jobid == "" {
+		return jobState, "", err
+	}
+	cmdPercentStr := fmt.Sprintf("azcopy jobs show %s | grep Percent", jobid)
+	// cmd out example:
+	// Percent Complete (approx): 100.0
+	summary, err := af.ExecCmd.RunCommand(cmdPercentStr)
+	if err != nil {
+		klog.Warningf("failed to get azcopy job with error: %v, jobState: %v", err, AzcopyJobError)
+		return AzcopyJobError, "", fmt.Errorf("couldn't show jobs summary in azcopy %v", err)
+	}
+	jobState, percent, err := parseAzcopyJobShow(summary)
+	if err != nil || jobState == AzcopyJobError {
+		klog.Warningf("failed to get azcopy job with error: %v, jobState: %v", err, jobState)
+		return AzcopyJobError, "", fmt.Errorf("couldn't parse azcopy job show in azcopy %v", err)
+	}
+	return jobState, percent, nil
+}
+
+func parseAzcopyJobList(joblist string, dstBlobContainer string) (string, AzcopyJobState, error) {
```

Comment:
pls add comments for all these functions in this file

Reply:
done